### PR TITLE
Date filter spec

### PR DESCRIPTION
**Pull request by adsantos**
This PR suggests a change in the spec structure by adding the property "freezeTime". FreezeTime should be used when running the tests by forcing the program to run with currentTime = freezeTime. This will allow to always get the same output results when running tests that are dependent on time, e.g. overdue tasks.
The PR also includes a spec for the "no due date" filter.

### CODE REVIEW

**Comment by adsantos** [comment]
Test data added

**Comment by adsantos** [comment]
Ready for review.

File: spec_no_due_date.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                "All whitespace is ignored and the query is case-insensitive.",
+                "The result is a single section with all the items without date date."
+            ],
+            "tests": [
+                {
+                    "id": "1",
+                    "freezeDate" : null,
+                    "currentUser" : null,
+                    "context": "When asked for tasks with no due date in input_overdue_no_due_date.json",
+                    "query": "overdue",
```
Comment (Garee) [inline spec_no_due_date.json]: I think you forgot to remove the `overdue` query from the `no due date` specification.

File: spec_overdue.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            ],
+            "tests": [
+                {
+                    "id": "1",
+                    "freezeDate" : "2015-07-13T18:25:43.511Z",
+                    "currentUser" : null,
+                    "context": "When asked for tasks that are overdue in input_overdue_no_due_date.json",
+                    "query": "overdue",
+                    "testFile": "input_overdue_no_due_date.json",
+                    "assertSections": [
```
Comment (Garee) [inline spec_overdue.json]: `overdue` should only be result in one section following yesterday's meeting.

File: input_overdue_no_due_date.json [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            "teamInboxProject" : false,
+            "itemOrder": 2
+        }
+    ],
+    "items": [
+        {
+            "id": 1,
+            "content": "Call David.",
+            "dueDate": "2015-07-11T18:25:43.511Z",
+            "priority": null,
```
Comment (Garee) [inline input_overdue_no_due_date.json]: Should we change `null` priorities to `4`?

File: spec_no_due_date.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                            "type": "Filtered",
+                            "priority": null,
+                            "date": null,
+                            "project" : null,
+                            "items": [4,5,6],
+                            "isReorderable": false,
+                            "labels": null
+                        }
+                    ]
+                }
```
Comment (Garee) [inline spec_no_due_date.json]: I think we should add an additional test for `noduedate` as whitespace is ignored.

**Comment by adsantos** [comment]
@Garee Ready for review again. I agree with all the comments you added and I've updated the json files accordingly.
